src/include/transaction/transaction_context.h

```diff
@@ -30,6 +31,28 @@ namespace terrier::transaction {
  */
 class TransactionContext {
  public:
+  class DebugLock : public std::shared_mutex {
```

Comment:
is this only used for debugging? If so, you can delete thiis after debugging.

Reply:
Yes - we will